specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-02-05/iotdps.json

```diff
@@ -0,0 +1,2558 @@
+            "description": "Name of the certificate to retrieve."
+          },
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Resource group identifier."
+          },
+          {
+            "name": "provisioningServiceName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Name of the provisioning service the certificate is associated with."
+          },
+          {
+            "name": "If-Match",
+            "in": "header",
+            "required": false,
+            "type": "string",
+            "description": "ETag of the certificate."
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Metadata for the specified certificate.",
+            "schema": {
+              "$ref": "#/definitions/CertificateResponse"
+            }
+          },
+          "default": {
+            "description": "Default error response.",
+            "schema": {
+              "$ref": "#/definitions/ErrorDetails"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "PUT"
+        ],
+        "summary": "Upload the certificate to the provisioning service.",
+        "description": "Add new certificate or update an existing certificate.",
+        "operationId": "DpsCertificate_CreateOrUpdate",
+        "x-ms-examples": {
+          "DPSCreateOrUpdateCertificate": {
+            "$ref": "./examples/DPSCertificateCreateOrUpdate.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "description": "Resource group identifier.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "provisioningServiceName",
+            "in": "path",
+            "description": "The name of the provisioning service.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "certificateName",
+            "in": "path",
+            "description": "The name of the certificate create or update.",
+            "required": true,
+            "type": "string",
+            "maxLength": 256
+          },
+          {
+            "name": "certificateDescription",
+            "in": "body",
+            "description": "The certificate body.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CertificateResponse"
+            }
+          },
```

Comment:
Request body changed from 2021-10-15 api-version
https://github.com/Azure/azure-rest-api-specs/blob/e495f3d2232ed4cd4850b6caf37febb21a88ef5f/specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2021-10-15/iotdps.json#L184-L192

This is a breaking change.

Reply:
Pass breaking changes review from Swagger does not mean you can break in Java.

Reply:
Reiterating our discussion offline, these changes are needed to repair the certificate create flow using ARM templates. Today the template is not compatible with our currently GA'd api.